crates/chain/src/local_chain.rs

```diff
@@ -85,8 +85,23 @@ impl LocalChain {
             .map(|&hash| BlockId { height, hash })
     }
 
-    /// This is like the sparsechain's logic, expect we must guarantee that all invalidated heights
-    /// are to be re-filled.
+    /// Get the most recent blocks of the best chain.
+    ///
+    /// `max_count` is the upper bound of the number of elements to return.
+    pub fn recent_blocks_of_best_chain(&self, max_count: usize) -> BTreeMap<u32, BlockHash> {
```

Comment:
Seems like another premature convenience method? It's not used in this PR so maybe wait until the PR where you actually want it to introduce it. The caller can do `.iter().rev().take(..)` for now.

I'm also not sure this is the *right* API. Perhaps this should take a height and give you all blocks >= that height and only the tip if that's empty. I don't know if you necessarily want `n` blocks because getting blocks that are 10,000 blocks deep already is a waste of time. I don't know the answer but I can't really decide here.

Reply:
The problem with taking a height is you can't be sure if the method will return any blocks at all, or return a block that you can connect to.

The benefit of `max_count` is that it is the lower bound of the "height range size" returned (so you can use a number that is large enough to guarantee no reorgs). It also acts as the upper bound of elements to clone.

In any case, I agree that this is premature. Will remove.